p2p/gossip/protocol.go

```diff
@@ -18,14 +18,51 @@ const propagateHandleBufferSize = 1000 // number of MessageValidation that we al
 const ProtocolName = "/p2p/1.0/gossip"
 const protocolVer = "0"
 
-type hash uint32
+type hash [12]byte
```

Comment:
I think we already have a definition for hash somewhere

Reply:
the hash we have it [32]byte hash and this hash is only 12 bytes. it is better that we store it in a [12]byte type. also this hash is only for internal use of this package so no worries of misuse.